src/SmartTransactionsController.ts

```diff
@@ -255,8 +289,34 @@ export default class SmartTransactionsController extends BaseController<
     return currentIndex === -1 || currentIndex === undefined;
   }
 
-  updateSmartTransaction(smartTransaction: SmartTransaction): void {
-    const { chainId } = this.config;
+  updateSmartTransaction(
+    smartTransaction: SmartTransaction,
+    { networkClientId }: { networkClientId?: NetworkClientId } = {},
+  ) {
+    let { chainId } = this.config;
+    let { ethQuery } = this;
+    if (networkClientId) {
+      const networkClient = this.getNetworkClientById(networkClientId);
+      chainId = networkClient.configuration.chainId;
+      ethQuery = new EthQuery(networkClient.provider);
+    }
```

Comment:
are these helpers useful here?

```
    const chainId = this.getChainId({ networkClientId });
    const ethQuery = this.getEthQuery({ networkClientId });
```

Reply:
Yeah perhaps. This was from a while ago but I think my thinking here was optimizing for fewer calls to `this.getNetworkClientById` but it doesn't really matter either way.